src/main/java/uk/gov/companieshouse/overseasentitiesapi/service/ManagingOfficerCessationService.java

```diff
@@ -0,0 +1,141 @@
+package uk.gov.companieshouse.overseasentitiesapi.service;
+
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.tuple.Pair;
+import org.springframework.stereotype.Service;
+import uk.gov.companieshouse.api.model.managingofficerdata.ManagingOfficerDataApi;
+import uk.gov.companieshouse.api.model.officers.CompanyOfficerApi;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.ManagingOfficerCorporateDto;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.ManagingOfficerIndividualDto;
+import uk.gov.companieshouse.overseasentitiesapi.model.dto.OverseasEntitySubmissionDto;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.cessations.Cessation;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.cessations.CorporateManagingOfficerCessation;
+import uk.gov.companieshouse.overseasentitiesapi.model.updatesubmission.changelist.cessations.IndividualManagingOfficerCessation;
+import uk.gov.companieshouse.overseasentitiesapi.utils.ApiLogger;
+
+@Service
+public class ManagingOfficerCessationService {
+
+  public static final String NO_PAIR_FOUND = "No matching MO was found in the database";
+  public static final String NO_ID_FOUND_IN_PRIVATE_DATA = "No Managing Officer ID was found in Private Data";
+  public static final String SERVICE = "ManagingOfficerCessationService";
+
+  public List<Cessation> managingOfficerCessations(
+      OverseasEntitySubmissionDto overseasEntitySubmissionDto,
+      Map<String, Pair<CompanyOfficerApi, ManagingOfficerDataApi>> combinedMoData,
+      Map<String, Object> logMap) {
+
+    List<Cessation> cessations = new ArrayList<>();
+    cessations.addAll(
+        getIndividualManagingOfficers(overseasEntitySubmissionDto, combinedMoData, logMap).stream()
+            .flatMap(Optional::stream)
+            .collect(Collectors.toList()));
+    cessations.addAll(
+        getCorporateManagingOfficersCessations(overseasEntitySubmissionDto, combinedMoData, logMap)
+            .stream()
+            .flatMap(Optional::stream)
+            .collect(Collectors.toList()));
+
+    return cessations;
+  }
+
+  private List<Optional<IndividualManagingOfficerCessation>> getIndividualManagingOfficers(
+      OverseasEntitySubmissionDto overseasEntitySubmissionDto,
+      Map<String, Pair<CompanyOfficerApi, ManagingOfficerDataApi>> combinedMoData,
+      Map<String, Object> logMap) {
+    var managingOfficersIndividual = overseasEntitySubmissionDto.getManagingOfficersIndividual();
+    return managingOfficersIndividual.stream()
```

Comment:
Can `managingOfficersIndividual` be null? I'm not sure if we're initialising this by default

Reply:
I think it (and other kinds of BO/MOs) are passed as [] in the model. Should be simple to test by running an update and only adding 1 individual BO say.